Traceback (most recent call last):
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/sc.uni-leipzig.de/le837wmue/.conda/envs/fin/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import xarray as xr
import fsspec
ds = xr.open_zarr(fsspec.get_mapper('http://data.rsc4earth.de/EarthSystemDataCube/v2.1.1/esdc-8d-0.25deg-1x720x1440-2.1.1.zarr/'), consolidated=True)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/registry.py:243[0m, in [0;36mget_filesystem_class[0;34m(protocol)[0m
[1;32m    242[0m [38;5;28;01mtry[39;00m:
[0;32m--> 243[0m     register_implementation(protocol, [43m_import_class[49m[43m([49m[43mbit[49m[43m[[49m[38;5;124;43m"[39;49m[38;5;124;43mclass[39;49m[38;5;124;43m"[39;49m[43m][49m[43m)[49m)
[1;32m    244[0m [38;5;28;01mexcept[39;00m [38;5;167;01mImportError[39;00m [38;5;28;01mas[39;00m e:

File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/registry.py:266[0m, in [0;36m_import_class[0;34m(cls, minv)[0m
[1;32m    265[0m mod, name [38;5;241m=[39m [38;5;28mcls[39m[38;5;241m.[39mrsplit([38;5;124m"[39m[38;5;124m.[39m[38;5;124m"[39m, [38;5;241m1[39m)
[0;32m--> 266[0m mod [38;5;241m=[39m [43mimportlib[49m[38;5;241;43m.[39;49m[43mimport_module[49m[43m([49m[43mmod[49m[43m)[49m
[1;32m    267[0m [38;5;28;01mreturn[39;00m [38;5;28mgetattr[39m(mod, name)

File [0;32m~/.conda/envs/rsapp/lib/python3.9/importlib/__init__.py:127[0m, in [0;36mimport_module[0;34m(name, package)[0m
[1;32m    126[0m         level [38;5;241m+[39m[38;5;241m=[39m [38;5;241m1[39m
[0;32m--> 127[0m [38;5;28;01mreturn[39;00m [43m_bootstrap[49m[38;5;241;43m.[39;49m[43m_gcd_import[49m[43m([49m[43mname[49m[43m[[49m[43mlevel[49m[43m:[49m[43m][49m[43m,[49m[43m [49m[43mpackage[49m[43m,[49m[43m [49m[43mlevel[49m[43m)[49m

File [0;32m<frozen importlib._bootstrap>:1030[0m, in [0;36m_gcd_import[0;34m(name, package, level)[0m

File [0;32m<frozen importlib._bootstrap>:1007[0m, in [0;36m_find_and_load[0;34m(name, import_)[0m

File [0;32m<frozen importlib._bootstrap>:986[0m, in [0;36m_find_and_load_unlocked[0;34m(name, import_)[0m

File [0;32m<frozen importlib._bootstrap>:680[0m, in [0;36m_load_unlocked[0;34m(spec)[0m

File [0;32m<frozen importlib._bootstrap_external>:850[0m, in [0;36mexec_module[0;34m(self, module)[0m

File [0;32m<frozen importlib._bootstrap>:228[0m, in [0;36m_call_with_frames_removed[0;34m(f, *args, **kwds)[0m

File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/implementations/http.py:11[0m, in [0;36m<module>[0;34m[0m
[1;32m      9[0m [38;5;28;01mfrom[39;00m [38;5;21;01murllib[39;00m[38;5;21;01m.[39;00m[38;5;21;01mparse[39;00m [38;5;28;01mimport[39;00m urlparse
[0;32m---> 11[0m [38;5;28;01mimport[39;00m [38;5;21;01maiohttp[39;00m
[1;32m     12[0m [38;5;28;01mimport[39;00m [38;5;21;01mrequests[39;00m

[0;31mModuleNotFoundError[0m: No module named 'aiohttp'

The above exception was the direct cause of the following exception:

[0;31mImportError[0m                               Traceback (most recent call last)
Input [0;32mIn [34][0m, in [0;36m<cell line: 3>[0;34m()[0m
[1;32m      1[0m [38;5;28;01mimport[39;00m [38;5;21;01mxarray[39;00m [38;5;28;01mas[39;00m [38;5;21;01mxr[39;00m
[1;32m      2[0m [38;5;28;01mimport[39;00m [38;5;21;01mfsspec[39;00m
[0;32m----> 3[0m ds [38;5;241m=[39m xr[38;5;241m.[39mopen_zarr([43mfsspec[49m[38;5;241;43m.[39;49m[43mget_mapper[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mhttp://data.rsc4earth.de/EarthSystemDataCube/v2.1.1/esdc-8d-0.25deg-1x720x1440-2.1.1.zarr/[39;49m[38;5;124;43m'[39;49m[43m)[49m, consolidated[38;5;241m=[39m[38;5;28;01mTrue[39;00m)

File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/mapping.py:230[0m, in [0;36mget_mapper[0;34m(url, check, create, missing_exceptions, alternate_root, **kwargs)[0m
[1;32m    199[0m [38;5;124;03m"""Create key-value interface for given URL and options[39;00m
[1;32m    200[0m 
[1;32m    201[0m [38;5;124;03mThe URL will be of the form "protocol://location" and point to the root[39;00m
[0;32m   (...)[0m
[1;32m    227[0m [38;5;124;03m``FSMap`` instance, the dict-like key-value store.[39;00m
[1;32m    228[0m [38;5;124;03m"""[39;00m
[1;32m    229[0m [38;5;66;03m# Removing protocol here - could defer to each open() on the backend[39;00m
[0;32m--> 230[0m fs, urlpath [38;5;241m=[39m [43murl_to_fs[49m[43m([49m[43murl[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    231[0m root [38;5;241m=[39m alternate_root [38;5;28;01mif[39;00m alternate_root [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m [38;5;28;01melse[39;00m urlpath
[1;32m    232[0m [38;5;28;01mreturn[39;00m FSMap(root, fs, check, create, missing_exceptions[38;5;241m=[39mmissing_exceptions)

File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/core.py:382[0m, in [0;36murl_to_fs[0;34m(url, **kwargs)[0m
[1;32m    380[0m [38;5;28;01melse[39;00m:
[1;32m    381[0m     protocol [38;5;241m=[39m split_protocol(url)[[38;5;241m0[39m]
[0;32m--> 382[0m     [38;5;28mcls[39m [38;5;241m=[39m [43mget_filesystem_class[49m[43m([49m[43mprotocol[49m[43m)[49m
[1;32m    384[0m     options [38;5;241m=[39m [38;5;28mcls[39m[38;5;241m.[39m_get_kwargs_from_urls(url)
[1;32m    385[0m     update_storage_options(options, kwargs)

File [0;32m~/.conda/envs/rsapp/lib/python3.9/site-packages/fsspec/registry.py:245[0m, in [0;36mget_filesystem_class[0;34m(protocol)[0m
[1;32m    243[0m         register_implementation(protocol, _import_class(bit[[38;5;124m"[39m[38;5;124mclass[39m[38;5;124m"[39m]))
[1;32m    244[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mImportError[39;00m [38;5;28;01mas[39;00m e:
[0;32m--> 245[0m         [38;5;28;01mraise[39;00m [38;5;167;01mImportError[39;00m(bit[[38;5;124m"[39m[38;5;124merr[39m[38;5;124m"[39m]) [38;5;28;01mfrom[39;00m [38;5;21;01me[39;00m
[1;32m    246[0m [38;5;28mcls[39m [38;5;241m=[39m registry[protocol]
[1;32m    247[0m [38;5;28;01mif[39;00m [38;5;28mgetattr[39m([38;5;28mcls[39m, [38;5;124m"[39m[38;5;124mprotocol[39m[38;5;124m"[39m, [38;5;28;01mNone[39;00m) [38;5;129;01min[39;00m ([38;5;124m"[39m[38;5;124mabstract[39m[38;5;124m"[39m, [38;5;28;01mNone[39;00m):

[0;31mImportError[0m: HTTPFileSystem requires "requests" and "aiohttp" to be installed
ImportError: HTTPFileSystem requires "requests" and "aiohttp" to be installed

